linalg/solvers.cpp

```diff
@@ -25,13 +25,18 @@ namespace mfem
 
 using namespace std;
 
+namespace detail
+{
+MFEM_DISABLE_WARNING_PUSH
+MFEM_DISABLE_WARNING_DEPRECATED
+}
+
```

Comment:
I'd remove the `namespace` and add a comment why we are disabling deprecated warnings here.

Reply:
I inroduced the hack with the namespace IIRC doxygen messes up the docs of the IterativeSolvers constructor, but yes, I should add a code comment with some details.

Reply:
Regarding the `namespace`: okay, I guess the lack of a semi-column at the end of the macro can confuse Doxygen. In other places we use `mfem::internal` (instead of `mfem::detail`) to hide things from Doxygen documentation, so we should either switch `detail` to `internal` or add `mfem::detail` here: https://github.com/mfem/mfem/blob/0843a87d7953cf23e556dcfd426d27bd9cfb3e21/doc/CodeDocumentation.conf.in#L857

Reply:
Thinking further about this, maybe an option to disable this suppression should be provided for debugging purposes, because silencing the warnings in the file can potentially mask other issues.

Reply:
Ideally, the deprecation warnings should be enabled just for the lines that generate such warnings. (My understanding is that we want to keep such lines for now, in order to preserve backward compatibility.)

Reply:
I pushed some changes that localizes the warning suppression only to places where `print_level` is modified.

Reply:
My changes worked with `clang` but not with `gcc`. ☹️

Reply:
I had to move some of the macros outside of the ctors -- I didn't find another way to make it work with GCC. This is probably good enough.

Reply:
Yes, that was my intention. Keeping the lines for backward compatibility, but enabling warnings to make users aware of the change. 
Also, thanks for taking the time here! I remember that, at the time of writing this hotfix, I also pulled it out of the macros, because I ran into some weird issues. But I agree that it should be good enoug now.